File: backend/WebApplication1/WebApplication1/Data/Assessment.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace WebApplication1.Data;

[Table("assessments")]
public class Assessment
{
    [Key]
    [Column("assessment_id")]
    public int AssessmentId { get; set; }

    [Column("resident_id")]
    public int ResidentId { get; set; }

    [Column("instrument")]
    public string Instrument { get; set; } = string.Empty;

    [Column("administered_date")]
    public DateOnly AdministeredDate { get; set; }

    [Column("administered_by")]
    public string AdministeredBy { get; set; } = string.Empty;

    [Column("total_score")]
    public decimal? TotalScore { get; set; }

    [Column("tick_count")]
    public int? TickCount { get; set; }

    [Column("severity_band")]
    public string? SeverityBand { get; set; }

    [Column("responses_json", TypeName = "jsonb")]
    public string? ResponsesJson { get; set; }

    [Column("worst_event")]
    public string? WorstEvent { get; set; }

    [Column("immediate_action")]
    public string? ImmediateAction { get; set; }

    [Column("observation_notes")]
    public string? ObservationNotes { get; set; }

    [Column("created_at")]
    public DateTime CreatedAt { get; set; }
}

File: frontend/intex/src/types/Assessment.ts
export type Instrument = 'PCL5' | 'SUICIDE_RISK' | 'BDI' | 'BAI'

export type Severity =
  | 'Minimal'
  | 'Mild'
  | 'Borderline'
  | 'Moderate'
  | 'Severe'
  | 'Extreme'
  | 'Low'
  | 'High'
  | 'Emergency'
  | 'NotPTSD'
  | 'ProbablePTSD'

export interface Assessment {
  assessmentId: number
  residentId: number
  instrument: Instrument
  administeredDate: string // YYYY-MM-DD
  administeredBy: string
  totalScore: number | null
  tickCount: number | null
  severityBand: Severity | null
  responsesJson: string | null
  worstEvent: string | null
  immediateAction: string | null
  observationNotes: string | null
  createdAt: string
}
